### DISCUSSION TITLE
How to disable compression

### DISCUSSION THREAD

**Discussion by Progman3K** [ImageMagick/ImageMagick#6501] [General] [answered]
Hi, 
I have an image file, which I am trying to save uncompressed.

I am using the command

convert test.bmp -define bmp:format=bmp5 -alpha on -compress none out.bmp

But for some reason, when I read in the image, it has compression set to type 3, not 0, in it's header:
Size of header: in code 40, on disk: 124
Dimensions: 234x333
planes: 1
bits per pixel: 32
compression: 3
size in bytes: 311688
colors used: 0
color important: 0

What am I doing wrong?

Thank you

**Comment by snibgo** [comment, marked as answer]
Are you doing anything wrong? What does "compression:3" mean?

234\*333\*4 = 311688, so there seems to be no compression.

**Reply by Progman3K** [reply]
> What does "compression:3" mean?

You're right, I took any value other than 0 to mean compression was used, but in fact _checks documentation_ the value 3 means BI_BITFIELDS, which is also uncompressed

Many thanks
